api/TextEncoder.json

```diff
@@ -70,86 +82,405 @@
             },
             "opera_android": {
               "version_added": null
             },
             "safari": {
-              "version_added": null
+              "version_added": false
             },
             "safari_ios": {
-              "version_added": null
+              "version_added": false
             },
             "samsunginternet_android": {
               "version_added": null
             }
           },
           "status": {
-            "experimental": false,
+            "experimental": true,
             "standard_track": true,
             "deprecated": false
           }
         }
       },
-      "encode": {
+      "TextEncoder": {
         "__compat": {
-          "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextEncoder/encode",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/TextEncoder/TextEncoder",
+          "description": "<code>TextEncoder()</code> constructor",
           "support": {
             "webview_android": {
+              "version_added": "38"
+            },
+            "chrome": [
+              {
+                "version_added": "53"
+              },
+              {
+                "version_added": "38",
+                "version_removed": "53",
+                "notes": "Throws <code>RangeError</code> exception for unknown encoding type"
+              }
+            ],
+            "chrome_android": {
+              "version_added": "38"
+            },
+            "edge": {
               "version_added": null
             },
-            "chrome": {
+            "edge_mobile": {
               "version_added": null
             },
-            "chrome_android": {
+            "firefox": [
+              {
+                "version_added": "48"
+              },
+              {
+                "version_added": "38",
+                "version_removed": "48",
+                "notes": "Invalid parameter leads to the standard <code>RangeError</code> exception."
+              },
+              {
+                "version_added": "19",
+                "version_removed": "38",
+                "notes": "<code>utf-16</code> not supported. Invalid parameter leads to a <code>TypeError</code> exception."
+              },
+              {
+                "version_added": "18",
+                "notes": "Slightly different version of the spec"
+              }
+            ],
+            "firefox_android": [
+              {
+                "version_added": "48"
+              },
+              {
+                "version_added": "38",
+                "version_removed": "48",
+                "notes": "Invalid parameter leads to the standard <code>RangeError</code> exception."
+              },
+              {
+                "version_added": "19",
+                "version_removed": "38",
+                "notes": "<code>utf-16</code> not supported. Invalid parameter leads to a <code>TypeError</code> exception."
```

Comment:
The MDN table says `As of Firefox 48, Chrome 54 and Opera 41, no other encoding types are allowed other than utf-8.`, so I'm not sure this is accurate. Shouldn't the note go with `version_added 48`?

Reply:
Good point. I overlooked that. Thanks!

Reply:
Just by reading the notes, it seems that firefox accepted a parameter on the constructor before v48, but not utf-16. That's ambiguous so I left it there but I added the note about forbidding parameters post-v48.

Reply:
Apparently [utf-16 was removed in v48](https://developer.mozilla.org/en-US/Firefox/Releases/48#Others) so that settles it, I suppose. I feel like a detective now haha